### ISSUE TITLE
`--verifytypes` reports missing annotation for `NewType` (Python 3.9 and earlier)

### ISSUE THREAD

**Issue by rsokl** [microsoft/pyright#4764]
Follow up from https://github.com/microsoft/pyright/discussions/4744

For a project consisting of: 

(edit: updated to import from `typing_extensions`, not `typing`)

```python
from typing_extensions import NewType

__all__ = ["Apple"]

Apple = NewType("Apple", int)
```

and for Python 3.9 and earlier, running  `--verifytypes` on the project raises 

```
error: Type is missing type annotation and could be inferred differently by type checkers
    Inferred type is "Type[Apple]
```

This error does not occur on Python 3.10.

[Here](https://github.com/rsokl/rsokl_pyright_example) is a minimal project to reproduce. 

<details>
  <summary>Repro:</summary>

```console
$ cd /tmp
$ git clone git@github.com:rsokl/rsokl_pyright_example.git
$ conda create -n py39 python=3.9
$ conda activate py39
$ pip install pyright rsokl_pyright_example/
$ pyright --verifytypes rsokl_pyright_example
Module name: "rsokl_pyright_example"
Package directory: "c:\Users\rsokl\pyright_bug_example\src\rsokl_pyright_example"
Module directory: "c:\Users\rsokl\pyright_bug_example\src\rsokl_pyright_example"
Path of py.typed file: "c:\Users\rsokl\pyright_bug_example\src\rsokl_pyright_example\py.typed"

Public modules: 1
   rsokl_pyright_example

Symbols used in public interface:
rsokl_pyright_example.Apple
  c:\Users\rsokl\pyright_bug_example\src\rsokl_pyright_example\__init__.py:5:1 - error: Type is missing type annotation and could be inferred differently by type checkers
    Inferred type is "Type[Apple]"

Symbols exported by "rsokl_pyright_example": 1
  With known type: 0
  With ambiguous type: 1
  With unknown type: 0
  Functions without docstring: 0
  Functions without default param: 0
  Classes without docstring: 0

Other symbols referenced but not exported by "rsokl_pyright_example": 0
  With known type: 0
  With ambiguous type: 0
  With unknown type: 0

Type completeness score: 0%

Completed in 0.566sec
```
</details>

**Comment by erictraut** [comment]
Yes, that makes sense. `typing.NewType` isn't available in Python 3.9 and older. You would need to specify 3.10 or switch to `typing_extensions.NewType` if you want this to verify without errors.

**Comment by rsokl** [comment]
`typing.NewType` has been available since Python 3.5.2, however I see now that it was changed from being a function to being a class in Python 3.10, which would explain why it is missing an annotation in earlier versions.

> or switch to typing_extensions.NewType if you want this to verify without errors.

This does not work. I updated the example repo and original post accordingly. 

That being being said, this is no longer an issue for me as I can run just run in Python 3.10 and I don't think this is a bug in pyright.

**Comment by erictraut** [comment]
Ah, you're right. I missed the fact that it changed from a function to a class. I'll need to add a bit more code to handle both cases.

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.299, which I just published.
